sdk/formrecognizer/Azure.AI.FormRecognizer/src/DocumentField.cs

```diff
@@ -31,23 +32,32 @@ internal DocumentField(SelectionMarkState? value, string content, IReadOnlyList<
         [CodeGenMember("Type")]
         public DocumentFieldType ValueType { get; }
```

Comment:
>DocumentFieldType [](http://example.com/codeflow?start=15&length=17)

You mentioned in the description that this approach has some drawbacks - curious what they are.

Reply:
My concerns is users would be inclined to attempt extracting the `InternalValue` object somehow. Not noticing it's private, some will try to find it in intellisense and think it's a bug when they can't find it. The reason we don't like exposing it as an `object` directly is because users usually struggle when trying to figure out which type they should cast to. That's why we introduced the `DocumentField` conversion methods (`AsInt64`, `AsDouble`, `AsAddress`, etc).